### PR TITLE
[cross] support arm-linux-gnueabihf cross compiling with netcore

### PR DESCRIPTION
**Pull request by lewurm**
Some notes on usage:

```console
$ sudo apt install g{++,cc}-gnu-linux-gnueabihf python3-pip

$ ./autogen.sh --disable-boehm --target=arm-linux-gnueabihf \
    --with-cross-offsets=arm-linux-gnueabihf-cross-offsets.h \
    --with-core=only \
    CFLAGS="-O0 -ggdb3 -fno-omit-frame-pointer" \
    CXXFLAGS="-O0 -ggdb3 -fno-omit-frame-pointer"

$ make -C tools/offsets-tool-py/

$ python3 tools/offsets-tool-py/offsets-tool.py \
    --targetdir=/home/lewurm/work/mono \
    --abi=arm-linux-gnueabihf \
    --monodir=/home/lewurm/work/mono \
    --outfile=arm-linux-gnueabihf-cross-offsets.h \
    --libclang=/usr/lib/x86_64-linux-gnu/libclang-7.so.1 \
    --sysroot=/usr/arm-linux-gnueabihf \
    --netcore
```

### CODE REVIEW

File: tools/offsets-tool-py/offsets-tool.py [hunk 1]
```diff
@@ -84,6 +86,14 @@ def require_emscipten_path (args):
 			self.target = Target ("TARGET_WASM", None, [])
 			self.target_args += ["-target", args.abi]
 
+		# Linux
+		elif "arm-linux-gnueabihf" == args.abi:
+			self.target = Target ("TARGET_ARM", None, ["ARM_FPU_VFP", "HAVE_ARMV5", "HAVE_ARMV6", "HAVE_ARMV7"] + LINUX_DEFINES)
+			self.target_args += ["--target=arm---gnueabihf"]
+			self.target_args += ["-I", "/usr/lib/gcc-cross/arm-linux-gnueabihf/8/include/"]
+			self.target_args += ["-I", "/usr/lib/gcc-cross/arm-linux-gnueabihf/8/include-fixed/"]
```
Comment (jaykrell) [inline tools/offsets-tool-py/offsets-tool.py]: Maybe `arm-linu-gnuabihf` on these lines, change to `args.abi`?

Comment (akoeplinger) [reply tools/offsets-tool-py/offsets-tool.py]: can we make these configurable too? on my Ubuntu 18.04 I only get `/usr/lib/gcc-cross/arm-linux-gnueabihf/7`

Comment (jaykrell) [reply tools/offsets-tool-py/offsets-tool.py]: Check if the directories exist?
If none at all, error?

Comment (jaykrell) [reply tools/offsets-tool-py/offsets-tool.py]: Try to generalize for any target? 
Gcc doesn’t do some of this itself?
Like, mainly just run the correct gcc?

Comment (lewurm) [reply tools/offsets-tool-py/offsets-tool.py]: @jaykrell this uses `clang`. Thanks for the comments, updated the PR.

Comment (jaykrell) [reply tools/offsets-tool-py/offsets-tool.py]: Sorry - clang doesn’t pick up these
Include directories itself? I’ll try later..

File: tools/offsets-tool-py/offsets-tool.py [hunk 2]
```diff
@@ -84,6 +87,33 @@ def require_emscipten_path (args):
 			self.target = Target ("TARGET_WASM", None, [])
 			self.target_args += ["-target", args.abi]
 
+		# Linux
+		elif "arm-linux-gnueabihf" == args.abi:
```
Comment (akoeplinger) [inline tools/offsets-tool-py/offsets-tool.py]: can we add `arm-linux-gnueabi` and  `aarch64-linux-gnu` too? this is awesome, thanks ❤️

**Comment by jaykrell** [comment]
Can it somehow generalize more automatically?

I’d like to just set CC and/or CXX and/or one or two extra parameters to configure (host & target; it sniffs uname or CC?) and make. 

Isn’t that usual model for cross build?
Asking too much?

File: tools/offsets-tool-py/offsets-tool.py (hunk 2 above)
Comment (lewurm) [reply tools/offsets-tool-py/offsets-tool.py]: I would suggest that we first get `arm-linux-gnueabihf` working on CI and then add the other targets. I'm sure there are bits missing :)

**Pull request merged**